src/lints/struct_added_without_non_exhaustive.ron

```diff
@@ -0,0 +1,57 @@
+SemverQuery(
+    id: "struct_added_without_non_exhaustive",
+    human_readable_name: "pub struct added without #[non_exhaustive]",
+    description: "A new public struct was added without #[non_exhaustive], making it harder to evolve in the future.",
+    required_update: Minor,
+    lint_level: Warn,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        name @output
+                        struct_type @output
+
+                        # Check that it's not marked #[non_exhaustive]
+                        attrs @filter(op: "not_contains", value: ["$non_exhaustive"])
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        # Verify all fields are public
+                        field @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            visibility_limit @filter(op: "!=", value: ["$public"])
+                        } 
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+            baseline @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                item {
+                    ... on Struct {
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
+        "true": true,
+        "non_exhaustive": "non_exhaustive",
```

Comment:
The `attrs` list contains attribute strings in their entirety, so the current filter will never match. This probably explains the false-positive you were seeing.
```suggestion
        "non_exhaustive": "#[non_exhaustive]",
```